addons/account/wizard/account_move_send.py

```diff
@@ -378,15 +398,18 @@ def _hook_invoice_document_before_pdf_report_render(self, invoice, invoice_data)
         return
 
     @api.model
-    def _prepare_invoice_pdf_report(self, invoice, invoice_data):
+    def _prepare_invoice_pdf_report(self, invoice, invoice_data, xml_id='account.account_invoices'):
         """ Prepare the pdf report for the invoice passed as parameter.
         :param invoice:         An account.move record.
         :param invoice_data:    The collected data for the invoice so far.
+        :param xml_id:          The xml_id of the invoice PDF template.
         """
         if invoice.invoice_pdf_report_id:
             return
 
-        content, _report_format = self.env['ir.actions.report']._render('account.account_invoices', invoice.ids)
+        if not xml_id:
+            raise UserError(_("Invoice PDF template should have xml_id"))
```

Comment:
The user error message is maybe a bit too technical, maybe we can change that, check with Christophe if we can find a better formulation or if we don't care :smile: 
Also maybe we could use a redirect warning here ? 

EDIT: With the passing of the argument in an api.model function maybe this is not a good way of dealing with the case that xml_id is not present

Las was proposing that you use the wizard parameter in the _process_send_and_print, maybe you can discuss with him to see what he has in mind